### ISSUE TITLE
Building VTK with HDF5 installed fails due to not linking zlib

### ISSUE THREAD

**Issue by Neumann-A** [Microsoft/vcpkg#4895]
VTK fails to build if HDF5 is installed because some subprojects do not correctly include linking zlib when USE_SYSTEM_HDF5 is on.

The subprojects are
* netcdf
* IOAMR

for netcdf the trick is to change the call to target_link_libraries to also link to vtkzlib_LIBRARIES
for ioamr the trick is to add vtkzlib as an additional dependency in the vtk module definition.

**Comment by Neumann-A** [comment]
problem is that an updated hdf5 port (1.10.4) is not properly propagating that linking to zlib is required.

**Comment by Neumann-A** [comment]
problem is in hdf5/src/CMakeLists.txt. line 1045ff and 1077ff

```
target_link_libraries (${HDF5_LIB_TARGET}
    PRIVATE ${LINK_LIBS} ${LINK_COMP_LIBS} "$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_C_LIBRARIES}>"
    PUBLIC $<$<NOT:$<PLATFORM_ID:Windows>>:${CMAKE_DL_LIBS}>
)
```
SZIP and ZLIB are added in hdf5/CMakeFilters.cmake to LINK_COMP_LIBS
I believe the PRIVATE was added in the updated version.

it would make sense to add (maybe even conditional if szip or ZLIB is used)
```
target_link_libraries (${HDF5_LIB_TARGET} PUBLIC szip-static ZLIB::ZLIB)
target_link_libraries (${HDF5_LIBSH_TARGET} PUBLIC szip-shared ZLIB::ZLIB)
```
 to hdf5/src/CMakeLists.txt

**Comment by luzpaz** [comment]
does an upstream hdf5 ticket need to be opened for this ?

**Comment by Neumann-A** [comment]
@luzpaz 
already did that see:
 https://forum.hdfgroup.org/t/cmake-zlib-dependency-not-properly-propagted-by-hdf5-targets/5149/2

Problem is why is hdf5-targets.cmake different for debug and release?

**Comment by Neumann-A** [comment]
closed by #4777
